redis/runtime/src/main/java/io/quarkiverse/langchain4j/redis/runtime/RedisEmbeddingStoreConfig.java

```diff
@@ -0,0 +1,83 @@
+package io.quarkiverse.langchain4j.redis.runtime;
+
+import static io.quarkus.runtime.annotations.ConfigPhase.BUILD_AND_RUN_TIME_FIXED;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkiverse.langchain4j.redis.MetricType;
+import io.quarkiverse.langchain4j.redis.VectorAlgorithm;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+/**
+ * Configuration of the Redis embedding store.
+ */
+@ConfigRoot(phase = BUILD_AND_RUN_TIME_FIXED)
+@ConfigMapping(prefix = "quarkus.langchain4j.redis")
+public interface RedisEmbeddingStoreConfig {
+
+    /**
+     * The name of the Redis client to use. These clients are configured by means of the `redis-client` extension.
+     * If unspecified, it will use the default Redis client.
+     */
+    @WithDefault("<default>")
+    String redisClientName();
+
+    /**
+     * The dimension of the embedding vectors. This has to be the same as the dimension of vectors produced by
+     * the embedding model that you use. For example, AllMiniLmL6V2QuantizedEmbeddingModel produces vectors of dimension 384.
+     * OpenAI's text-embedding-ada-002 produces vectors of dimension 1536.
+     *
+     */
+    Integer dimension();
```

Comment:
Shouldn't this be a long?

Reply:
Don't think that there are any embedding models producing such huge vectors, but yeah why not